identity-account/src/identity/identity_state.rs

```diff
@@ -45,7 +49,39 @@ use crate::types::KeyLocation;
 type Properties = VerifiableProperties<BaseProperties>;
 type BaseDocument = CoreDocument<Properties, Object, Object>;
 
-pub type RemoteEd25519<'a, T> = JcsEd25519<RemoteSign<'a, T>>;
+pub struct RemoteEd25519;
+
+impl Named for RemoteEd25519 {
+  const NAME: &'static str = "JcsEd25519Signature2020";
+}
+
+impl RemoteEd25519 {
+  pub async fn create_signature<U, T>(data: &mut U, method: impl Into<String>, secret: &RemoteKey<'_, T>) -> Result<()>
+  where
+    U: Serialize + SetSignature,
+    T: Storage,
+  {
+    data.set_signature(Signature::new(Self::NAME, method));
+
+    let value: SignatureValue = Self::sign(&data, secret).await?;
+    let write: &mut Signature = data.try_signature_mut()?;
+
+    write.set_value(value);
+
+    Ok(())
+  }
+
+  pub async fn sign<T, X>(data: &X, remote_key: &RemoteKey<'_, T>) -> Result<SignatureValue>
+  where
+    X: Serialize,
+    T: Storage,
+  {
+    let message: Vec<u8> = data.to_jcs()?;
+    let signature: Vec<u8> = RemoteSign::sign(&message, remote_key).await?;
+    let signature: String = encode_b58(&signature);
+    Ok(SignatureValue::Signature(signature))
```

Comment:
Discussion: to prevent the complete duplication of fundamental signing operations we could split them up into steps in the `Sign` trait and re-use them rather than copy-paste? E.g. have a `prepare_data` function which converts it to JCS, a `generate_signature` function which actually signs it, and a default `sign` method which calls both of those.

This may be overcomplicating it and doesn't really solve the duplication issue, it just enables some re-use of the code from `JcsEd25519`. Though since so little code is actually duplicated I think it's fine to leave as-is.